Lib/test/test_types.py

```diff
@@ -713,6 +713,14 @@ def test_or_type_repr(self):
         assert repr(int | None) == "int | None"
         assert repr(int | typing.GenericAlias(list, int)) == "int | list[int]"
 
+    def test_annotation_name(self):
+        an = types.AnnotationName("x")
+        self.assertEqual(repr(an), "<unresolved name: x>")
+        # TODO for some reason __name__ doesn't exist if I don't do this first
```

Comment:
This is really weird to me: somehow `.__name__` only exists on AnnotationName objects after I've done something like `AnnotationName.__dict__`. My implementation mostly followed that of `GenericAlias`, which doesn't have this problem. I'll stare at it some more, but maybe someone who knows more about CPython internals will know the reason.